parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -72,6 +75,27 @@ propagateAndApply patch branch = do
   f     <- applyPropagate edits
   pure $ Branch.step (f . applyDeprecations patch) branch
 
+-- Creates a mapping from old data constructors to new data constructors
+-- by looking at the original names for the data constructors which are
+-- embedded in the Decl object because we carefully planned that.
+generateConstructorMapping
```

Comment:
This uses the names in the original `Decl` to figure out which constructors of the old type should get replaced with which constructors of the new type. This lets us reassign those names to constructors of the new type.